apps/hash-api/src/graph/knowledge/system-types/file.ts

```diff
@@ -87,26 +144,24 @@ export const createFileFromUploadRequest: ImpureGraphFunction<
 };
 
 export const createFileFromExternalUrl: ImpureGraphFunction<
-  Omit<CreateEntityParams, "properties" | "entityTypeId"> &
-    MutationCreateFileFromUrlArgs,
+  MutationCreateFileFromUrlArgs,
   Promise<File>
 > = async (ctx, authentication, params) => {
-  const { ownedById, description, entityTypeId, url } = params;
+  const { description, displayName, url } = params;
 
   const filename = url.split("/").pop()!;
-  const mimeType = mime.lookup(filename) || "application/octet-stream";
 
-  const fileEntityTypeId =
-    // @todo validate that entityTypeId, if provided, ultimately inherits from File
-    //    (or checks that it has at least the same properties)
-    entityTypeId || types.entityType.file.entityTypeId;
+  const { entityTypeId, existingEntity, mimeType, ownedById } =
+    await generateCommonParameters(ctx, authentication, params, filename);
 
   try {
     const properties: FileProperties = {
       "https://blockprotocol.org/@blockprotocol/types/property-type/description/":
         description ?? undefined,
       "https://blockprotocol.org/@blockprotocol/types/property-type/file-name/":
         filename,
```

Comment:
We might want to add these to the `ontology-types.ts` file at some point, so they can be re-used more easily

Reply:
I have already added these to `libs/@local/hash-subgraph/src/main.ts` (maybe not the best place for them, but it's where `linkEntityTypeUrl` already was). In this instance it isn't necessary to use those aliases though, since they are keys on `FileProperties` and therefore we have autocomplete and type safety already.